### PR TITLE
Expand the RemotePlaybackAvailability description

### PR DESCRIPTION
**Pull request by avayvod**
Add algorithms for monitoring available remote playback devices.

### CODE REVIEW

**Comment by avayvod** [comment]
@mounirlamouri PTaL

File: index.html [hunk 1]
```diff
@@ -333,12 +333,240 @@
...
+          The <a>user agent</a> MUST keep track of the <dfn>set of
+          availability objects</dfn> requested through the <code><a for=
+          "RemotePlayback">getAvailability</a>()</code> method. The
+          <a>set of availability objects</a> is represented as a set of
+          tuples <em>(<var>A</var>, <var>availabilitySource</var>)</em>,
+          initially empty, where:
+        </p>
+        <ol>
+          <li>
+            <var>A</var> is a live <a>RemotePlaybackAvailability</a> object;
```
Comment (mounirlamouri) [inline index.html]: s/A/availabilityObject/ ?

File: index.html [hunk 2]
```diff
@@ -333,12 +333,240 @@
...
-        <b>Draft summary:</b><br>
-        The <code><a>RemotePlaybackAvailability</a></code> interface exposes
-        whether there is a device on the user's network able to play one of the
-        sources available for the associated media element. The <code>change</code>
-        event is fired each time when the <code>value</code> changes.
+        A <a><code>RemotePlaybackAvailability</code></a> object is associated with available <a>remote playback devices</a>
+        and represents the <dfn>remote playback device availability</dfn> for the media element. If the user
+        agent can <a>monitor the list of available remote playback devices</a> in the background (without
+        a pending request to <code><a for="RemotePlayback">connect()</a></code>), the 
+        <a><code>RemotePlaybackAvailability</a></code> object MUST be implemented by the user agent.
```
Comment (mounirlamouri) [inline index.html]: What if it doesn't support?

File: index.html [hunk 3]
```diff
@@ -333,12 +333,240 @@
...
+        <p>
+          While there are live <a>RemotePlaybackAvailability</a> objects, the
+          <a>user agent</a> MAY <a>monitor the list of available remote playback devices</a> continuously,
+          so that pages can use the <a for="RemotePlaybackAvailability">value</a> property of a
+          <a>RemotePlaybackAvailability</a> object to offer remote playback only
+          when there are available devices. However, the <a>user agent</a>
+          may not support continuous availability monitoring; for example,
+          because of platform or power consumption restrictions. In this case
+          the <a>Promise</a> returned by <code><a for="RemotePlayback">getAvailability</a>()</code>
+          MUST be <a data-lt="reject">rejected</a> and the algorithm to <a>monitor
```
Comment (mounirlamouri) [inline index.html]: rejected with what?

File: index.html [hunk 4]
```diff
@@ -333,12 +333,240 @@
...
+        </dl>
+        <ol>
+          <li>Let <var>P</var> be a new <a>Promise</a>.
+          </li>
+          <li>Return <var>P</var>, but continue running these steps <a>in
+          parallel</a>.
+          </li>
+          <li>If the user agent is unable to <a>monitor the list of available
+          remote playback devices</a> for the entire lifetime of the browsing context
+          (for instance, because the user has disabled this feature), then:
```
Comment (mounirlamouri) [inline index.html:452]: We should consider killing this.

File: index.html [hunk 5]
```diff
@@ -333,12 +333,240 @@
...
+          </dt>
+          <dd>
+            <var>P</var>, a <a>Promise</a>
+          </dd>
+        </dl>
+        <ol>
+          <li>Let <var>P</var> be a new <a>Promise</a>.
+          </li>
+          <li>Return <var>P</var>, but continue running these steps <a>in
+          parallel</a>.
```
Comment (mounirlamouri) [inline index.html]: Can you move this after the early return cases? and make them _return_ a promise?

File: index.html [hunk 6]
```diff
@@ -333,12 +333,240 @@
...
+              <li>
+                <a>Resolve</a> <var>P</var> with <var>A</var>.
+              </li>
+              <li>
+                Abort all the remaining steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            Let <var>A</var> be a new <code>RemotePlaybackAvailability</code>
```
Comment (mounirlamouri) [inline index.html]: ditto, let's use something else that `A`

File: index.html [hunk 7]
```diff
@@ -333,12 +333,240 @@
...
+        </h4>
+        <p>
+          If the <a>set of availability objects</a> is non-empty, or there is
+          a pending request to <a for="RemotePlayback" data-lt="connect">initiate remote playback</a>,
+          the <a>user agent</a> MUST <dfn>monitor the list of available remote playback devices</dfn> by
+          running the following steps:
+        </p>
+        <ol link-for="RemotePlaybackAvailability">
+          <li>
+            Retrieve available remote playback devices (using an implementation specific mechanism)
```
Comment (mounirlamouri) [inline index.html:519]: implementation and system

**Comment by mounirlamouri** [comment]
lgtm with comments. It's a bit hard to follow everything in a diff but we can easily iterate on it if some things need to be adjusted.

File: index.html (hunk 4 above)
Comment (avayvod) [inline index.html:452]: Could you elaborate a little more? Do you think this passage is excessive as part of the algorithm? Or do you think we shouldn't reject the promise in this case?

File: index.html (hunk 5 above)
Comment (avayvod) [inline index.html]: I think the Promise should be returned by the algorithm as early as possible to allow async operations. If the early return check like user gesture can be done in a sync manner, fine, but I don't think it should be mandated by the algorithm.

File: index.html (hunk 6 above)
Comment (avayvod) [inline index.html]: Done. I liked short names that stand out on its own though because of a capital letter even if CSS fails.

File: index.html (hunk 7 above)
Comment (avayvod) [inline index.html:519]: I think implementation is flexible enough and covers system as well.

File: index.html (hunk 3 above)
Comment (avayvod) [inline index.html]: Done.

File: index.html (hunk 2 above)
Comment (avayvod) [inline index.html]: There're two places where the spec says the promise is rejected then. Though it may be worth repeating for self-containment. Done.

File: index.html (hunk 1 above)
Comment (avayvod) [inline index.html]: Done.
